### PR TITLE
Replace use of boost::asio::deadline_timer with boost::asio::steady_timer

### PR DESCRIPTION
**Pull request by jonesmz**

### CODE REVIEW

File: include/mqtt/client.hpp [hunk 1]
```diff
@@ -356,25 +356,25 @@ class client : public endpoint<std::mutex, std::lock_guard, PacketIdBytes> {
      * See https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc464635115<BR>
      * 3.1.2.10 Keep Alive
      */
-    void set_keep_alive_sec_ping_ms(std::uint16_t keep_alive_sec, std::size_t ping_ms) {
-        if (ping_duration_ms_ != 0 && base::connected() && ping_ms == 0) {
+    void set_keep_alive_sec_ping_ms(std::uint16_t keep_alive_sec, std::chrono::steady_clock::duration ping) {
```
Comment (redboltz) [inline include/mqtt/client.hpp:359]: I agree with providing `std::chrono::steady_clock::duration ping` parameter version. But this function should be kept as the original for compatibility.

I think that adding the parameter `std::chrono::steady_clock::duration ping` to the following function's 2nd parameter is better.

```cpp
void set_keep_alive_sec(std::uint16_t keep_alive_sec) {
```

And the default value of the parameter is `keep_alive_sec/2`.

Comment (redboltz) [reply include/mqtt/client.hpp:359]: I mean something like as follows:

```cpp
void set_keep_alive_sec_ping_ms(std::uint16_t keep_alive_sec, std::size_t ping_ms) {
    // call set_keep_alive_sec()
}

void set_keep_alive_sec(std::uint16_t keep_alive_sec,  std::chrono::steady_clock::duration ping = /* keep_alive_sec/2 */) {
    // set timer implementation
}
```

Comment (jonesmz) [reply include/mqtt/client.hpp:360]: Changed

Comment (redboltz) [reply include/mqtt/client.hpp:360]: At first, I didn't understand why you don't use default parameter. But now I understand that the default parameter can't refer to other parameter.
So overload is reasonable solution. 

In the name of function `set_keep_alive_sec_ping_ms()`, ms means milliseconds. 

```cpp
    void set_keep_alive_sec_ping_ms(std::uint16_t keep_alive_sec, std::chrono::steady_clock::duration ping);
    void set_keep_alive_sec_ping_ms(std::uint16_t keep_alive_sec, std::size_t ping_ms);
    void set_keep_alive_sec(std::uint16_t keep_alive_sec);
```

Now we have three keep alive setting functions. The first one, user can use not only ms but also any `std::chrono::steady_clock::duration`. So the name of function should be set_keep_alive_sec().

So the three functions will become as follows:

```cpp
    // you added
    void set_keep_alive_sec(std::uint16_t keep_alive_sec, std::chrono::steady_clock::duration ping);
    // original functions
    void set_keep_alive_sec_ping_ms(std::uint16_t keep_alive_sec, std::size_t ping_ms);
    void set_keep_alive_sec(std::uint16_t keep_alive_sec);
```

Could you update the PR?


P.S. 
Sorry for my late response. I live in Tokyo. I'm fune but due to COVID-19, I need to manage my lifeline. So I need prioritize many things.

Comment (jonesmz) [reply include/mqtt/client.hpp:360]: No problem.

Please stay healthy.

No urgency of reviews.

Comment (jonesmz) [reply include/mqtt/client.hpp]: Change made

**Comment by redboltz** [comment]
Thank you for updating. megerd.

**Pull request merged**